### ISSUE TITLE
na_ontap_autosupport not idempotent with proxy authentication

### ISSUE THREAD

**Issue by mamoep** [ansible-collections/netapp#83]
##### SUMMARY

The module na_ontap_autosupport always reports "changed" when a proxy_url with authentication is used.

##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME

na_ontap_autosupport

##### ANSIBLE VERSION

```paste below
ansible 2.9.15
  config file = /ansible/private/ansible.cfg
  configured module search path = [u'/home/username/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python2.7/site-packages/ansible
  executable location = /usr/bin/ansible
  python version = 2.7.5 (default, Aug 13 2020, 02:51:10) [GCC 4.8.5 20150623 (Red Hat 4.8.5-39)]

```

##### CONFIGURATION

```paste below

```

##### OS / ENVIRONMENT

RHEL 7.9
Collection 21.2.0
ONTAP 9.7P8

##### STEPS TO REPRODUCE

Execute the play

```yaml
  - na_ontap_autosupport:
      state: present
      node_name: cluster-01
      transport: https
      mail_hosts: mailhost.domain.local
      from_address: user@domain.local
      to_addresses: user@domain.local
      proxy_url: USER:PASS@proxyhost.domain.local
      hostname: cluster.domain.local
      <<: *login
    connection: local

```

##### EXPECTED RESULTS

Result ok when configuration is unchanged.

`ok: [cluster.domain.local]`

##### ACTUAL RESULTS

Result is always changed.
 

```paste below
changed: [cluster.domain.local]
```

**Comment by carchi8py** [comment]
We have an internal story (3649), open for this.

**Comment by lonico** [comment]
"Fixed" in ONTAP collection 21.4.0

na_ontap_autosupport - warn when password is present in proxy_url as it makes the operation not idempotent.

If a password is present, we cannot compare its value to the current value, as it is obfuscated.
